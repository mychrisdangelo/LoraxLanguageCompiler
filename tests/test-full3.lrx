/* 
 * Author: Chris D'Angelo
 * End to end test of vanilla assignment and tree literals
 */

int main()
{
	tree <char>t(2);
	tree <int>t2(3);
	tree <float>t3(3);
	tree <bool>t4(2);
	t = 'h'['i'['j', 'o'], 'k'];
	t2 = 1[2, 3[4, 5, 6], 7];
	t3 = 1.0[2.1, 3.1, 4.1[5.2, 5.3, 5.4]];
	t4 = true[false[true, false], true];
	print(t, t2, t3, t4);
}