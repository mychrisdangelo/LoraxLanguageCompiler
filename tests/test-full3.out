h[i[j[null,null],o[null,null]],k[null,null]]1[2[null,null,null],3[4[null,null,null],5[null,null,null],6[null,null,null]],7[null,null,null]]1.000000[2.100000[null,null,null],3.100000[null,null,null],4.100000[5.200000[null,null,null],5.300000[null,null,null],5.400000[null,null,null]]]true[false[true[null,null],false[null,null]],true[null,null]]